src/Behat/Behat/Context/Snippet/Generator/ContextSnippetGenerator.php

```diff
@@ -197,30 +180,29 @@ private function getUsedClasses(StepNode $step)
     /**
      * Generates snippet template using regex, method name and arguments.
      *
-     * @param string   $pattern
-     * @param string   $methodName
      * @param string[] $methodArguments
-     *
-     * @return string
      */
-    private function getSnippetTemplate($pattern, $methodName, array $methodArguments)
+    private function getSnippetTemplate(string $pattern, string $methodName, array $methodArguments): string
     {
         return sprintf(
-            self::$templateTemplate,
-            str_replace('%', '%%', $pattern),
+            self::$snippetTemplate,
+            $this->preparePattern($pattern),
             $methodName,
             implode(', ', $methodArguments)
         );
     }
 
+    private function preparePattern(string $pattern): string
+    {
+        $pattern = str_replace('%', '%%', $pattern);
+        $pattern = str_replace('\\\\', '\\\\\\\\', $pattern);
```

Comment:
instead of those replacements, I suggest removing the quotes from the template and using `var_export()` to make PHP generate the proper literal representation of the string with any needed escaping.

Reply:
You suggest something like that?

```php
        $pattern = str_replace('%', '%%', $pattern);

        return var_export($pattern, true);
```

4 tests from snippets.feature fail with this change, since some `\` are now escaped too. I presume it's normal and expected.

Reply:
@fmatsos I'm guessing that is what @stof meant, it looks ok to me. Could you clarify about the tests - is it that snippets.feature is currently valid and using var_export gives the wrong result? Or is it that the current expectations in snippets.feature are not correctly escaped and using var_export fixes that? Thanks

Reply:
Of course! I'll test the final regex to check if new pattern is valid or not.

Reply:
Hi @stof 

I check deeply your suggestion, but it seems I can't use `var_export` in this case. If I use it, I have `\` from regex keywords that are escaped too. So the pattern is not valid anymore.

Example:
- expected: `#[Then('/^do something undefined with \\(\d+)$/')]`
- result: `#[Then(''/^do something undefined with \\\\(\\d+)$/'')]`